ui/lib/kv/addon/components/kv-secret-form.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+import KVObject from 'vault/lib/kv-object';
+
+/**
+ * @module KvSecretForm is used for creating a new secret or secret version (also considered 'editing')
+ *
+ * <KvSecretForm
+ *  @secret={{@secret}}
+ *  @onSave={{transition-to "vault.cluster.secrets.backend.kv.secret.details" @secret.path}}
+ *  @onCancel={{transition-to "vault.cluster.secrets.backend.kv.list"}}
+ * />
+ *
+ * @param {model} secret - Ember data model: 'kv/data'
+ * @param {callback} onSave - callback (usually a transition) from parent to perform after the model is saved
+ * @param {callback} onCancel - callback (usually a transition) from parent to perform when cancel button is clicked
+ */
+
+export default class KvSecretForm extends Component {
+  @service flashMessages;
+  @tracked showJsonView = false;
+  @tracked errorMessage;
+  @tracked modelValidations;
+  @tracked invalidFormAlert;
+
+  get emptyJson() {
+    // if secretData is null, this specially formats a blank object and renders a nice initial state for the json editor
+    return KVObject.create({ content: [{ name: '', value: '' }] }).toJSONString(true);
+  }
+
+  @action
+  toggleJsonView() {
+    this.showJsonView = !this.showJsonView;
+  }
+
+  @action
+  handleJson(value, codemirror) {
+    codemirror.performLint();
+    const lintingErrors = codemirror.state.lint.marked.length > 0;
+    if (!lintingErrors) {
+      this.args.secret.secretData = JSON.parse(value);
```

Comment:
As this is, linting errors don't block the user from submitting the form, which shows a value different than what we will submit. Probably an edge case but something to consider, not a blocker

Reply:
Going to slack message you as I'm actually contemplating a similar issue now - with the initial empty object being different than the actual value of `secretData` and how to handle this